mesonbuild/mcompile.py

```diff
@@ -356,6 +356,33 @@ def add_arguments(parser: 'argparse.ArgumentParser') -> None:
         help='Arguments to pass to `xcodebuild` (applied only on `xcode` backend).'
     )
 
+# Used by mdevenv.py
+def compile_single_executable(environment: Environment, target: T.Optional[str]) -> T.Tuple[T.Optional[str], int]:
```

Comment:
Surely, this function belongs in mdevenv.py then.

Reply:
It's using all the logic to convert the target name to a backend specific command, not worth moving all that into a shared module.